### PR TITLE
pre-commit improvements

### PR DESCRIPTION
**Pull request by kse3hi**
# Description

Disable performing pre-commit on save of the file. 
Instead pre-commit will be performed before commit.
Extension for pre-commit was removed.

Note: if developer doesn't want to recreate devContainer, he should perform `sudo pre-commit install`

## Azure DevOps PBI/Task reference

AB#_11724

## Checklist

* [x] Vehicle App can be started with dapr run and is connecting to vehicle data broker
* [x] Vehicle App can process MQTT messages and call the seat service
* [x] Vehicle App can be deployed to local K3D and is running
* [x] Created/updated tests, if necessary. Code Coverage percentage on new code shall be >= 70%.
* [x] Extended the documentation in Velocitas repo
* [x] Extended the documentation in README.md

* [x] Devcontainer can be opened successfully
* [x] Devcontainer can be opened successfully behind a corporate proxy
* [x] Devcontainer can be re-built successfully

* [x] Release workflow is passing

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -45,7 +45,6 @@
 		"matangover.mypy",
 		"anweber.vscode-httpyac",
 		"augustocdias.tasks-shell-input",
-		"elagil.pre-commit-helper",
```
Comment (kse3hi) [inline .devcontainer/devcontainer.json:48]: We don't use any functionality from this extension

File: .devcontainer/scripts/postCreateCommand.sh [hunk 2]
```diff
@@ -45,3 +47,9 @@ pip3 install setuptools==59.6.0
 # dependency to python-Levenshtein
 # wheels are missing and have to built from scratch
 sudo apt-get update && sudo apt-get install -y build-essential python3-dev
+
+# make sure that git is setup properly
+git config --global --add safe.directory /workspaces/vehicle-app-python-template
```
Comment (kse3hi) [inline .devcontainer/scripts/postCreateCommand.sh:52]: Fix for known issue :octocat:

Comment (MP91) [reply .devcontainer/scripts/postCreateCommand.sh:52]: this won't work if someone uses the template with a different namen

File: app/requirements.txt [hunk 3]
```diff
@@ -6,25 +6,25 @@
...
     # via -r app/requirements.in
 deprecation==2.1.0
     # via cloudevents
-frozenlist==1.3.1
+frozenlist==1.3.3
     # via
     #   aiohttp
     #   aiosignal
-grpcio==1.49.1
+grpcio==1.48.2
```
Comment (MP91) [inline app/requirements.txt:27]: why did you update dapr and grpcio?

File: .vscode/tasks.json [hunk 4]
```diff
@@ -12,7 +12,7 @@
 			"httpPort": 3500,
 			"type": "dapr",
 			"presentation": {
-				"close": true,
+				"close": false,
```
Comment (MP91) [inline .vscode/tasks.json:15]: is this needed?

File: NOTICE-3RD-PARTY-CONTENT.md [hunk 5]
```diff
@@ -4,68 +4,59 @@
...
 |distlib|0.3.6|Python Software Foundation License|
 |filelock|3.8.0|Public Domain|
 |flake8|5.0.4|MIT|
-|frozenlist|1.3.1|Apache 2.0|
+|frozenlist|1.3.3|Apache 2.0|
 |gitdb|4.0.9|BSD|
 |GitPython|3.1.29|BSD|
 |grpc-stubs|1.24.11|MIT|
-|grpcio|1.49.1|Apache 2.0|
+|grpcio|1.48.2|Apache 2.0|
```
Comment (MP91) [inline NOTICE-3RD-PARTY-CONTENT.md:26]: why did you downgrade grpcio?

File: NOTICE-3RD-PARTY-CONTENT.md [hunk 6]
```diff
@@ -4,68 +4,59 @@
...
 |black|22.10.0|MIT|
-|build|0.8.0|MIT|
+|build|0.9.0|MIT|
 |cfgv|3.3.1|MIT|
 |charset-normalizer|2.1.1|MIT|
 |click|8.1.3|New BSD|
-|cloudevents|1.6.1|Apache 2.0|
-|dapr|1.7.0|Apache 2.0|
+|cloudevents|1.7.1|Apache 2.0|
+|dapr|1.8.3|Apache 2.0|
```
Comment (MP91) [inline NOTICE-3RD-PARTY-CONTENT.md:17]: why do you use an untested version of dapr?

File: .devcontainer/devcontainer.json [hunk 7]
```diff
@@ -11,12 +11,14 @@
 	"settings": {
 		"python.pythonPath": "/usr/bin/python3",
 		"python.defaultInterpreterPath": "/usr/bin/python3",
+		// Only Flake8 is used as linter and static code analyzer, as faster tool
 		"python.linting.enabled": true,
-		"python.linting.pylintEnabled": true,
-		"python.linting.pylintPath": "/usr/local/bin/pylint",
+		"python.linting.pylintEnabled": false,
```
Comment (MP91) [inline .devcontainer/devcontainer.json:16]: is this needed if we don't have pylint at all?

File: NOTICE-3RD-PARTY-CONTENT.md (hunk 5 above)
Comment (kse3hi) [reply NOTICE-3RD-PARTY-CONTENT.md:26]: It is made by aligning dapr and protobuf by pip-tools

**Comment by kse3hi** [comment]
- [ ] Extend contribution.md
- [ ] remove .pylintrc / some settings pylint

File: .devcontainer/scripts/container-set.sh [hunk 8]
```diff
@@ -20,4 +20,4 @@ useradd vscode --password vscode -m
 usermod -aG sudo vscode
 
 sudo chown -R vscode $HOME
-echo "export PATH=\"$PATH:/root/.local/bin/\"" >> /home/vscode/.profile
+echo "export PATH=\"$PATH:/root/.local/bin:$HOME/.local/bin\"" >> /home/vscode/.profile
```
Comment (MP91) [inline .devcontainer/scripts/container-set.sh:23]: do you really need this?

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -25,9 +25,9 @@ https://www.eclipse.org/projects/handbook/#resources-commit
 * Use [Black](https://black.readthedocs.io/) to format your code.
 * Use [isort](https://isort.readthedocs.io/) to sort imports.
 * Use [pydocstyle](https://pydocstyle.readthedocs.io/) to check for PEP-8 style issues.
+* Use [bundit](https://pypi.org/project/bandit/) to check security issues.
```
Comment (MP91) [inline CONTRIBUTING.md:28]:
```suggestion
* Use [bandit](https://pypi.org/project/bandit/) to check security issues.
```

File: .devcontainer/scripts/container-set.sh (hunk 8 above)
Comment (kse3hi) [reply .devcontainer/scripts/container-set.sh:23]: Yes. WIthout this you need to specify for VSCODE task additional options

File: .vscode/scripts/runtime/local/pre-commit-run.sh [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+echo "#######################################################"
+echo "### Pre Commit                                      ###"
+echo "#######################################################"
+
+pre-commit run --show-diff-on-failure --color=always --all-files
```
[the remark this answers is not in the document]
Comment (kse3hi) [reply .vscode/scripts/runtime/local/pre-commit-run.sh:20]: If you will have some issue it is only way for debug VSCODE task

File: .devcontainer/scripts/container-set.sh (hunk 8 above)
Comment (MP91) [reply .devcontainer/scripts/container-set.sh:23]: but it is already in the path...

File: .vscode/scripts/runtime/local/pre-commit-run.sh (hunk 10 above)
[the remark this answers is not in the document]
Comment (MP91) [reply .vscode/scripts/runtime/local/pre-commit-run.sh:20]: why do we need to debug a task? is this a real use case? 

I would like to remove this one-line-file... it is even not part of the local runtime

File: .vscode/tasks.json [hunk 11]
```diff
@@ -99,6 +99,14 @@
 			"group": "none",
 			"isBackground": false,
 		},
+		{
+			"label": "Local - Pre Commit Action",
+			"detail": "Perform checking/formating of the code",
+			"type": "shell",
+			"command": "./.vscode/scripts/runtime/local/pre-commit-run.sh",
```
Comment (MP91) [inline .vscode/tasks.json:106]: please use the command directly

File: .github/workflows/ci.yml [hunk 12]
```diff
@@ -64,7 +66,7 @@ jobs:
       - name: unit test
         shell: bash
         run: |
-          pip install -r app/tests/requirements.txt
+          pip install -r app/tests/requirements-tests.txt
```
Comment (MP91) [inline .github/workflows/ci.yml:69]: do we really need to install this twice?

**Review by MP91 — CHANGES_REQUESTED** [review]

**Review by MP91 — APPROVED** [review]
LGTM

**Pull request merged**